source/coap_connection_handler.c

```diff
@@ -266,7 +266,10 @@ static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephem
 
         // Set socket option to receive packet info
         socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_RECVPKTINFO, &(const bool) {1}, sizeof(bool));
-
+        if (socket_interface_selection != -1) {
+            // Select socket interface if selection requested
+            socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_INTERFACE_SELECT, &socket_interface_selection, sizeof(int8_t));
```

Comment:
@kjbracey-arm 
This is meant for Multicast message sendind so that it will go to ethernet only. 
LinkLocal multicast goes to specific interface?
Sitelocal multicast goes to all interfaces? is multicast filtering rules checked?